compiler/aarch64/codegen/FPTreeEvaluator.cpp

```diff
@@ -90,11 +90,12 @@ OMR::ARM64::TreeEvaluator::fconstEvaluator(TR::Node *node, TR::CodeGenerator *cg
 
    union {
       float f;
-      int32_t i;
+      double d;
    } fvalue;
 
    fvalue.f = node->getFloat();
-   loadConstant32(cg, node, fvalue.i, tmpReg);
+   uint32_t imm = getFPImm(fvalue.d);
+   loadFPSinglePrecision(cg, node, imm, tmpReg);
```

Comment:
What do you do when `imm` is -1?
Consider the case where the float value cannot be represented by the 8-bit modified immediate.

Reply:
You don't need to generate the `fmov_wtos` instruction when the float value is represented by the 8-bit modified immediate.